src/server/models/pullRequestsModel.ts

```diff
@@ -6,22 +6,77 @@ const PR_COLLECTION_NAME = 'pull-requests';
 type Status = 'error'|'failure'|'pending'|'success';
 
 export type PRDetails = {
-  commits: {[key: string]: CommitDetails}; automerge: AutomergeOpts | undefined;
+  commits?: {[key: string]: CommitDetails};
+  automerge?: AutomergeOpts;
+  open?: boolean;
 };
 
 export type CommitDetails = {
   status: Status;
 };
 
 class PullRequestsModel {
-  async setCommitStatus(prId: string, commitId: string, status: Status) {
-    const prDoc = await firestore().collection(PR_COLLECTION_NAME).doc(prId);
+  private getDocID(owner: string, repo: string, num: number) {
+    if (!owner) {
+      throw new Error('No owner.');
+    }
+
+    if (!repo) {
+      throw new Error('No repo.');
+    }
+
+    if (!num) {
+      throw new Error('No num.');
+    }
```

Comment:
Num can't be null or empty? This would rule out 0 which may be ok, but probably unnecessary.